samples/matter/lock/src/zcl_callbacks.cpp

```diff
@@ -23,11 +23,17 @@ void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &a
 	VerifyOrReturn(attributePath.mClusterId == DoorLock::Id &&
 		       attributePath.mAttributeId == DoorLock::Attributes::LockState::Id);
 
+	/* Post events only if current lock state is different than given */
 	switch (*value) {
 	case to_underlying(DlLockState::kLocked):
-		GetAppTask().PostEvent(AppEvent(AppEvent::Lock, BoltLockManager::OperationSource::kRemote));
+		if (!BoltLockMgr().IsLocked()) {
```

Comment:
Sorry for back and forth, but I think it's already checked inside `BoltLockManager::Lock` :):
```
VerifyOrReturn(mState != State::kLockingCompleted);
```
So maybe nothing is actually needed. But we can clean this up later if you're tired of tesing it :)

Reply:
Yes I know, but this check is done after handling an event that was posted before, so it will not avoid unnecessary posts.

Reply:
Oh, I misunderstood your previous question. I meant that posting an extra event is not a big deal. But nevermind, both solutions work for me.